File: pidev-3A4-oussama/src/com/esprit/veltun/model/Maintenance.java
package com.esprit.veltun.model;

import java.sql.Date;
import com.esprit.veltun.model.base.BaseEntity;

public class Maintenance extends BaseEntity {
    private int id_demande;
    private String description;
    private String status;
    private Date submission_date;
    private int rating ;


    private int id_reclamation;

    public Maintenance() {
    }
    public Maintenance(String description, String status,Date submission_date,int rating) {
        this.description = description;
        this.status = status;
        this.submission_date = submission_date;
        this.rating =rating;
    }
    public Maintenance(String description, String status,Date submission_date) {
        this.description = description;
        this.status = status;
        this.submission_date = submission_date;
    }

    public Maintenance(int id_demande ,String description, String status,Date submission_date,int id_reclamation) {
        this.id_demande = id_demande;
        this.description = description;
        this.status = status;
        this.submission_date = submission_date;
        this.id_reclamation= id_reclamation;
    }

    public int getId_demande() {
        return id_demande;
    }

    public void setId_demande(int id_demande) {
        this.id_demande = id_demande;
    }
    public int getRating() {
        return rating;
    }

    public void setRating(int rating) {
        this.rating = rating;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }
    public String getStatus() {
        return status;
    }

    public void setStatus(String status) {
        this.status = status;
    }
    public Date getSubmission_date() {
        return submission_date;
    }

    public void setSubmission_date(Date submission_date) {
        this.submission_date = submission_date;
    }

    public int getId_reclamation() {
        return id_reclamation;
    }

    public void setId_reclamation(int id_reclamation) {
        this.id_reclamation = id_reclamation;
    }

    @Override
    public String toString() {
        return "maintenance [ id_demande=" + id_demande + ", description=" + description
                + ", status=" + status + ", submission_date=" + submission_date + ",  id_reclamation=" + id_reclamation + ", rating=" + rating +"]";
    }


}

File: webProject/src/Entity/Maintenance.php
<?php

namespace App\Entity;

use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Validator\Constraints as Assert;

#[ORM\Entity]
class Maintenance
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private ?int $idDemande = null;

    #[ORM\Column(type: 'text')]
    #[Assert\NotBlank]
    private ?string $description = null;

    #[ORM\Column(length: 20)]
    #[Assert\Choice(choices: ['en cours', 'resolu'], message: 'Invalid status.')]
    private ?string $status = 'en cours';

    #[ORM\Column(type: 'datetime')]
    private ?\DateTime $submissionDate = null;

    #[ORM\Column(type: 'integer')]
    #[Assert\Choice(choices: [1, 2, 3, 4, 5], message: 'Invalid rating.')]
    private ?int $rating = null;

    public function getIdDemande(): ?int
    {
        return $this->idDemande;
    }

    public function setIdDemande(?int $idDemande): void
    {
        $this->idDemande = $idDemande;
    }

    public function getDescription(): ?string
    {
        return $this->description;
    }

    public function setDescription(?string $description): void
    {
        $this->description = $description;
    }

    public function getStatus(): ?string
    {
        return $this->status;
    }

    public function setStatus(?string $status): void
    {
        $this->status = $status;
    }

    public function getSubmissionDate(): ?\DateTime
    {
        return $this->submissionDate;
    }

    public function setSubmissionDate(?\DateTime $submissionDate): void
    {
        $this->submissionDate = $submissionDate;
    }

    public function getRating(): ?int
    {
        return $this->rating;
    }

    public function setRating(?int $rating): void
    {
        $this->rating = $rating;
    }
    private $cin;
    public function getCin(): \User
    {
        return $this->cin;
    }
    public function setCin(\User $cin): Maintenance
    {
        $this->cin = $cin;
        return $this;
    }
}